api/python/tests/integration/test_packages.py

```diff
@@ -0,0 +1,57 @@
+""" Integration tests for T4 Packages. """
+import jsonlines
+import os
+import pytest
+
+from t4 import Package, PhysicalKeyType
+
+from mock import patch
+
+LOCAL_MANIFEST = os.path.join(os.path.dirname(__file__), 'data', 'local_manifest.jsonl')
+REMOTE_MANIFEST = os.path.join(os.path.dirname(__file__), 'data', 't4_manifest.jsonl')
+
+def mock_make_api_call(operation_name):
+    """ Mock boto3's AWS API Calls for testing. """
+    if operation_name == 'GetObject':
+        parsed_response = {'Body': {'foo'}}
+        return parsed_response
+    raise NotImplementedError
+
+def test_read_manifest(tmpdir):
+    """ Verify reading serialized manifest from disk. """
+    pkg = Package.load(open(LOCAL_MANIFEST))
+
+
+    with pytest.raises(NotImplementedError):
+        pkg.get('foo')
+
+    out_path = os.path.join(tmpdir, 'new_manifest.jsonl')
+    pkg.dump(open(out_path, "w"))
+    
+    # Insepct the jsonl to verify everything is maintained, i.e.
+    # that load/dump results in an equivalent set.
+    original_set = list(jsonlines.Reader(open(LOCAL_MANIFEST)))
```

Comment:
Wouldn't it be easier to just `Package.load` the second file and compare the contents of the packages? That would verify that write/read works without depending on the internals of it.

Reply:
Added a todo for now, needs the __eq__ to make this easier to work with.

Reply:
Ok. Can you use `with open` here and below, though?